Rego/Utils/DefenderUtils.rego

```diff
@@ -115,4 +119,45 @@ SensitiveAccounts(SensitiveAccountsSetting, SensitiveAccountsConfig) := true if
         SensitiveAccountsSetting.ExcludedDomains == SensitiveAccountsConfig.ExcludedDomains
     ]
     count([x | x := Mismatches[_]; x == false]) == 0
+}
+
+##############################################
+# Impersonation protection support functions #
+##############################################
+
+ImpersonationProtectionSetting(Policies, IdentityString) := Policy if {
+    Policy := [
+        Policy | Policy := Policies[_];
+        regex.match(IdentityString, Policy.Identity) == true;
+        Policy.Enabled == true;
+        Policy.EnableTargetedUserProtection == true
+    ]
+} else := set()
+
+ImpersonationProtectionConfig(PolicyID) := IncludedUsers if {
+    SensitiveUsers := input.scuba_config.Defender[PolicyID].SensitiveAccounts
+    IncludedUsers := { lower(trim_space(x)) | some x in SensitiveUsers.IncludedUsers; x != null }
+} else := set()
+
+ImpersonationProtection(Policies, IdentityString, IncludedUsers) := {
+    "Result" : true,
+    "Policy" : {
+        "Name" : Policy[0].Identity,
+        "Users" : PolicyProtectedUsers,
+        "Action" : Policy[0].TargetedUserProtectionAction
+    }
+} if {
+    Policy := ImpersonationProtectionSetting(Policies, IdentityString)
+    count(Policy[0]) > 0
+
+    PolicyProtectedUsers := { lower(x) | x := Policy[0].TargetedUsersToProtect[_] }
+    count(PolicyProtectedUsers) > 0
```

Comment:
Do we also want to assert that the config file is not empty? (e.g., `count(IncludedUsers) > 0`). Maybe...but I could go either way on this.

Reply:
That is true, before it was `count(PolicyProtectedUsers - IncludedUsers) == 0` so I only needed to check `PolicyProtectedUsers `, but I had to swap them so it would still pass if the settings had more users indicated than the config.